### PR TITLE
Add support for metric variants

### PR DESCRIPTION
**Pull request by davecromberge**
Note this pull request is a continuation of and depends on: dalmatinerdb/dqe_idx#11

### CODE REVIEW

**Comment by Licenser** [comment]
`dqe_idx 0.1.20` includes the required changes

File: src/query_builder.erl [hunk 1]
```diff
@@ -38,7 +39,7 @@ build_lookup_query(Collection, Metric, Grouping) ->
     {ok, Query, Values}.
 
 build_lookup_query(Bucket, Metric, Where, Grouping)
-  when is_list(Metric) ->
+  when is_list(Metric); is_atom(Metric) ->
```
Comment (Licenser) [inline src/query_builder.erl]: Curious why the is_atom here?

Comment (davecromberge) [reply src/query_builder.erl]: Metric is `undefined` in the case that the `select all from collection` query construction was used.  The immediate alternative I can think of is:

```
   when is_list(Metric); Metric =:= undefined
```

I did explore other approaches in terms of different tagged tuples {in... vs {all..., but this approach was cleanest.

Comment (Licenser) [reply src/query_builder.erl]: I think ; `Metric =:= undefined` is clearer, also guards against people passing un `undfined` like I'll frequently do!

Comment (Licenser) [reply src/query_builder.erl]: I saw that in a second place too I guess that's the same guard for `undefiend`?

**Comment by Licenser** [comment]
awesome!

File: src/query_builder.erl (hunk 1 above)
Comment (davecromberge) [reply src/query_builder.erl]: Haha you mispelled it again just to prove the point!

**Comment by Licenser** [comment]
published as `0.2.14`

**Comment by davecromberge** [comment]
Thanks Heinz!

**Pull request merged**